cirq/protocols/json_test.py

```diff
@@ -444,7 +446,14 @@ def test_mutually_exclusive_blacklist():
 ]
 
 
-@pytest.mark.parametrize('cirq_type,cls', _get_all_public_classes())
+def _roundtrip_test_classes():
+    for cirq_type, cls in _get_all_public_classes():
+        yield cirq_type, cls
+    yield ('_NamedConstantXmonDevice',
+           cirq.google.known_devices._NamedConstantXmonDevice)
```

Comment:
Interesting... so these things aren't exposed at the top-level namespace because `cirq.google` is not "flattened". 

Instead of this other function, you could add the yield to the end of `_get_all_public_classes`. There's already a couple ad-hoc items there. 

The real solution would be: refactor `_get_public_classes` to take a module object to extract public classes from and then chain together `_get_public_classes(cirq) + _get_public_classes(cirq.google)`. You'd also have to go through and add the resultant test failures to `SHOULDNT_BE_SERIALIZED` or `NOT_YET_SERIALIZABLE`.

If you want, you can do the second thing as part of this PR. Otherwise, open an issue and link to it in a comment with the ad-hoc insertion of `_NamedConstantXmonDevice`

Reply:
As a final thing: the actual objects returned by this inspecting function would be `Foxtail` and `Bristlecone` instead of `_NamedConstantXmonDevice`. This should still provide full test coverage of what we care about

Reply:
That still wouldn't get `_NamedConstantXmonDevice` because it is not public in cirq.google (and shouldn't be), though, so we'd still need the ad-hoc insertion, right?

Reply:
Right, only `Foxtail` and `Bristlecone` would show up, but those would take care of testing `_NamedConstantXMonDevice`. 

It's like how we test for `UNCONSTRAINED_DEVICE` which is a singleton of type `_UnconstrainedDevice`

Reply:
I've refactored the tests. I still have this other function to handle the ad-hoc cases like complex, np.ndarray, etc.